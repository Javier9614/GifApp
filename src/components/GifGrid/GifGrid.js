import React from 'react';
import GifGridItem from './GifGridItem/GifGridItem';
import {useFetchGifs} from '../../hooks/useFetchGifs';
import PropTypes from "prop-types";

import "./GifGrid.css";


const GifGrid = ({category}) => {


//aqui estamos cambiando el campo data a images para que se entienda mejor el codigo
const {data:images, loading} = useFetchGifs(category);

    return(

        <>
        <h4 className="animate__animated animate__fadeInDown">{category}</h4>

        {loading && <p className="animate__animated animate__flash">Loading</p> }
        
        <div className="card-grid">
            
                {
                    images.map(img =>(
                        <GifGridItem 
                        key={img.id}
                        //aqui destructuramos img para mandarle al hijo 
                        //data,url y title
                        {...img }/>
                    ))
                }
        </div>
        </>
            )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired,
}

export default GifGrid;