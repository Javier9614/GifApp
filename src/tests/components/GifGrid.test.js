import React from 'react';
import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';

//Esto va a simular la llamada a la api para el test
jest.mock("../../hooks/useFetchGifs")

describe('Pruebas en <GifGrid>',()=>{

    
    const category = "Bleach";
    test("Debe mostrar <GifGrid/> correctamente", ()=>{
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true,
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();

    })

    test("Debe mostrar items cuando se cargan gifs UseFetchGifs",()=>{

        const gifs = [{
            id: '123',
            url: 'http://localhost/bgdyhsua.gif',
            title:'Cualquier cosa'
        },
        {
            id: '321',
            url: 'http://localhost/bgdyhsua.gif',
            title:'Cualquier cosa'
        }
    ]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false,
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").lenght).toBe(gifs.lenght);
        

    })
})