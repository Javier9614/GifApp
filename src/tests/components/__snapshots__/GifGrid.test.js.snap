// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid> Debe mostrar <GifGrid/> correctamente 1`] = `
<Fragment>
  <h4
    className="animate__animated animate__fadeInDown"
  >
    Bleach
  </h4>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
